src/sbd-inquisitor.c

```diff
@@ -1307,37 +1303,50 @@ int main(int argc, char **argv, char **envp)
         } else if (strcmp(argv[optind], "ping") == 0) {
             exit_status = ping_via_slots(argv[optind + 1], servants_leader);
 
+        } else
+#endif
+        if (strcmp(argv[optind], "query-watchdog") == 0) {
+            exit_status = watchdog_info();
+        } else if (strcmp(argv[optind], "test-watchdog") == 0) {
+            exit_status = watchdog_test();
         } else if (strcmp(argv[optind], "watch") == 0) {
+            /* sleep $(sbd $SBD_DEVICE_ARGS dump | grep -m 1 msgwait | awk '{print $4}') 2>/dev/null */
+
+                const char *delay_source = "watchdog-timeout * 2";
+
+#if SUPPORT_SHARED_DISK
                 if(disk_count > 0) {
                     /* If no devices are specified, its not an error to be unable to find one */
                     open_any_device(servants_leader);
                 }
 
-                if (delay_start) {
-                    if (delay <= 0) {
-                        delay = get_first_msgwait(servants_leader);
-                    }
+                if (delay_start && delay <= 0) {
+                    delay = get_first_msgwait(servants_leader);
 
-                    sleep((unsigned long) delay);
+                    if (delay > 0) {
+                        delay_source = "msgwait";
+                    }
                 }
-
-	} else {
-		exit_status = -2;
-	}
 #endif
+                /* Re-calculate timeout_watchdog_warn based on any timeout_watchdog from:
+                 * SBD_WATCHDOG_TIMEOUT, -1 option or on-disk setting read with open_any_device() */
+                if (do_calculate_timeout_watchdog_warn) {
+                    timeout_watchdog_warn = calculate_timeout_watchdog_warn(timeout_watchdog);
+                }
 
-        /* Re-calculate timeout_watchdog_warn based on any timeout_watchdog from:
-         * SBD_WATCHDOG_TIMEOUT, -1 option or on-disk setting read with open_any_device() */
-        if (do_calculate_timeout_watchdog_warn) {
-            timeout_watchdog_warn = calculate_timeout_watchdog_warn(timeout_watchdog);
-        }
+                if (delay_start) {
+                    if (disk_count == 0 && delay <= 0) {
+                        delay = 2 * timeout_watchdog;
+                        delay_source = "SBD_WATCHDOG_TIMEOUT * 2";
```

Comment:
we don't know where watchdog-timeout is coming from - yet.
so we probably should stay with what delay_source already has.